src/index.js

```diff
@@ -27,12 +27,18 @@ export default declare(({
     SVG_DEFAULT_PROPS_CODE,
   }) => {
     const namedTemplate = `
-      var SVG_NAME = function SVG_NAME(props) { return SVG_CODE; };
+      var SVG_NAME = function SVG_NAME(props) { React.PureComponent.call(this, props); };
+      SVG_NAME.prototype = Object.create(React.PureComponent.prototype);
+      SVG_NAME.prototype.constructor = SVG_NAME;
+      SVG_NAME.prototype.render = function render() { var props = this.props; return SVG_CODE; };
```

Comment:
this seems very strange; only class components are pure.

Reply:
wdym? whats expected here? 🤔 want me to change it to `React.Component`?

Reply:
afaik this is a class component, only thing its not doing its being strict about how babel compiles a class to a function (eg, doing constructor call checks)

Reply:
oh.. nice, react big monorepo without docs with the only docs linked being broken..

I'll assume this is the same createReactClass and rely on old docs and use it then

Reply:
@ljharb, I've made the switch, can you check the current changes now?